### PR TITLE
change service quotas check logic in GCR as QuotaService is not avail…

### PR DESCRIPTION
**Pull request by hanrobyn**
As the QuotaService is not available in GCR, so I measure quotas for RDS and ELB using services itself. 

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: layers/utils/src/main/java/com/amazon/aws/partners/saasfactory/saasboost/Utils.java [hunk 1]
```diff
@@ -141,6 +141,17 @@ public static String toJson(Object obj) {
...
+    public static boolean isCnRegion(final String region) {
+        return region.startsWith("cn-");
+    }
+
+    public static String getUrlSuffix(final String region) {
+        if (isCnRegion(region)) {
+            return "amazonaws.com.cn";
+        }
+        return "amazonaws.com";
+    }
```
Comment (PoeppingT) [inline layers/utils/src/main/java/com/amazon/aws/partners/saasfactory/saasboost/Utils.java:153]: Why not use the `isChinaRegion(String)` and `endpointSuffix(String)` functions already in this file?

File: services/quotas-service/src/main/java/com/amazon/aws/partners/saasfactory/saasboost/QuotasServiceDAL.java [hunk 2]
```diff
@@ -367,6 +369,77 @@ private Map<String, Double> getQuotas(String serviceCode) {
...
+    private Map<String, Double> getRdsInstancesQuota() {
+        long instances;
+        long clusters;
+        try {
+            clusters = rds.describeAccountAttributes().accountQuotas().stream()
+                    .filter(quota -> quota.accountQuotaName().equals("DBClusters"))
+                    .findFirst().map(AccountQuota::max).orElse(40L);
+            instances = rds.describeAccountAttributes().accountQuotas().stream()
+                    .filter(quota -> quota.accountQuotaName().equals("DBInstances"))
+                    .findFirst().map(AccountQuota::max).orElse(40L);
```
Comment (PoeppingT) [inline services/quotas-service/src/main/java/com/amazon/aws/partners/saasfactory/saasboost/QuotasServiceDAL.java:415]: Let's cache the result of `rds.describeAccountAttributes().accountQuotas` so we don't waste time doing a second round-trip call to RDS.

File: services/quotas-service/src/main/java/com/amazon/aws/partners/saasfactory/saasboost/QuotasServiceDAL.java [hunk 3]
```diff
@@ -367,6 +369,77 @@ private Map<String, Double> getQuotas(String serviceCode) {
...
+        return retVals;
+    }
+
+    private Map<String, Double> getELBQuota() {
+        long instances;
+        try {
+            instances = elb.describeAccountLimits().limits().stream()
+                    .filter(x -> x.name().equals("application-load-balancers"))
+                    .findFirst().map(Limit::max).map(Long::valueOf).orElse(50L);
+        } catch (SdkServiceException rdsError) {
```
Comment (PoeppingT) [inline services/quotas-service/src/main/java/com/amazon/aws/partners/saasfactory/saasboost/QuotasServiceDAL.java:433]: small suggestion: call this `elbError`

**Review by PoeppingT — CHANGES_REQUESTED** [review]

File: layers/utils/src/main/java/com/amazon/aws/partners/saasfactory/saasboost/Utils.java (hunk 1 above)
Comment (hanrobyn) [reply layers/utils/src/main/java/com/amazon/aws/partners/saasfactory/saasboost/Utils.java]: change the code to use isChinaRegion(String) and endpointSuffix(String) functions

File: services/quotas-service/src/main/java/com/amazon/aws/partners/saasfactory/saasboost/QuotasServiceDAL.java (hunk 2 above)
Comment (hanrobyn) [reply services/quotas-service/src/main/java/com/amazon/aws/partners/saasfactory/saasboost/QuotasServiceDAL.java:415]: I think if we cache this result, then account changes its quota, we may use the old value.

File: services/quotas-service/src/main/java/com/amazon/aws/partners/saasfactory/saasboost/QuotasServiceDAL.java (hunk 3 above)
Comment (hanrobyn) [reply services/quotas-service/src/main/java/com/amazon/aws/partners/saasfactory/saasboost/QuotasServiceDAL.java]: I have renamed it.

File: services/quotas-service/src/main/java/com/amazon/aws/partners/saasfactory/saasboost/QuotasServiceDAL.java (hunk 2 above)
Comment (hanrobyn) [reply services/quotas-service/src/main/java/com/amazon/aws/partners/saasfactory/saasboost/QuotasServiceDAL.java]: I use a accountQuotas field to cache the result.

**Review by PoeppingT — APPROVED** [review]
LGTM, thanks for the submission!

**Pull request merged**